### PR TITLE
docs: add transparent proxy visual aid

### PR DESCRIPTION
**Pull request by boxofrad**
Adds a diagram (and accompanying text) demonstrating the before and after of Transparent Proxy.

### CODE REVIEW

File: website/content/docs/connect/transparent-proxy.mdx [hunk 1]
```diff
@@ -9,6 +9,35 @@ description: |-
 
 # Transparent Proxy
 
+Transparent proxy allows applications to communicate through the mesh without changing their configuration.
+It also hardens application security by preventing direct inbound connections that bypass the mesh.
```
Comment (karl-cardenas-coding) [inline website/content/docs/connect/transparent-proxy.mdx:13]:
```suggestion
Transparent proxy also hardens application security by preventing direct inbound connections that bypass the mesh.
```

File: website/content/docs/connect/transparent-proxy.mdx [hunk 2]
```diff
@@ -9,6 +9,35 @@ description: |-
...
+#### With Transparent Proxy
+
+![Diagram demonstrating that with transparent proxy, connections are automatically routed through the mesh](/img/consul-connect/with-transparent-proxy.png)
+
+With transparent proxy:
+
+1. Upstreams are inferred from service intentions, so no explicit configuration
+   is needed.
+1. Outbound connections to KubeDNS names "just work" — rules redirect them
+   through the proxy.
```
Comment (karl-cardenas-coding) [inline website/content/docs/connect/transparent-proxy.mdx:35]:
```suggestion
1. Outbound connections pointing to a KubeDNS names resolve and connect. Internal rules redirect them through the proxy.
```
I'm trying to make this statement seem less magical. The rules part could probably use a better description.

**Review by karl-cardenas-coding — APPROVED** [review]
Minor suggestions. Otherwise, this is great content.

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/569423.

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit c1cb58bdcbafcf1db8eb43dddf8eb592d09c4fbe onto `stable-website` succeeded!

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit c1cb58bdcbafcf1db8eb43dddf8eb592d09c4fbe onto `release/1.11.x` succeeded!

**Pull request merged**